avm/res/devops-infrastructure/managed-devops-pools/tests/e2e/max/main.tests.bicep

```diff
@@ -0,0 +1,146 @@
+
+@description('Optional. A token to inject into the name of each resource.')
+param namePrefix string = '#_namePrefix_#'
+
+// ========= //
+// Variables //
+// ========= //
+var azureDevOpsOrganizationName = 'contoso'
+var azureDevOpsProjectName = 'ContosoProject'
+
+// ============ //
+// Dependencies //
+// ============ //
+module nestedDependencies 'dependencies.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, resourceLocation)}-nestedDependencies'
+  params: {
+    devCenterName: 'dep-${namePrefix}-dc-${serviceShort}'
+    devCenterProjectName: 'dep-${namePrefix}-dcp-${serviceShort}'
+    managedIdentityName: 'dep-${namePrefix}-msi-${serviceShort}'
+    virtualNetworkName: 'dep-${namePrefix}-vnet-${serviceShort}'
+  }
+}
+
+// General resources
+// =================
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+  name: resourceGroupName
+  location: resourceLocation
+}
+
+// ============== //
+// Test Execution //
+// ============== //
+
+@batchSize(1)
+module testDeployment '../../../main.bicep' = [
+  for iteration in ['init', 'idem']: {
+    scope: resourceGroup
+    name: '${uniqueString(deployment().name, resourceLocation)}-test-${serviceShort}-${iteration}'
+    params: {
+      name: '${namePrefix}${serviceShort}001'
+      location: resourceLocation
+      agentProfile: {
+        kind: 'Stateless'
+        resourcePredictions: {
+          timeZone: 'Central Europe Standard Time'
+          daysData: [
+            {
+              '09:00:00': 1
+              '17:00:00': 0
+            }
+            {}
+            {}
+            {}
+            {}
+            {}
+            {}
+          ]
+        }
+        resourcePredictionsProfile: {
+          kind: 'Automatic'
+          predictionPreference: 'Balanced'
+        }
+      }
+      concurrency: 1
+      devCenterProjectResourceId: nestedDependencies.outputs.devCenterProjectId
+      images: [
+        {
+          aliases: [
+            'windows-2022'
+          ]
+          buffer: '*'
+          wellKnownImageName: 'windows-2022/latest'
+        }
+      ]
+      fabricProfileSkuName: 'Standard_DS2_v2'
+      organizationProfile: {
+        kind: 'AzureDevOps'
+        organizations: [
+          {
+            url: 'https://dev.azure.com/${azureDevOpsOrganizationName}'
+            parallelism: 1
+            projects: [
+              azureDevOpsProjectName
+            ]
+          }
+        ]
+        permissionProfile: {
+          kind: 'CreatorOnly'
+        }
+      }
+      dataDisks: [
+        {
+          caching: 'ReadWrite'
+          diskSizeGiB: 100
+          driveLetter: 'B'
+          storageAccountType: 'Standard_LRS'
+        }
+      ]
+      logonType: 'Interactive'
+      subnetId: nestedDependencies.outputs.subnetId
+      osDiskStorageAccount: 'Standard'
+      secretsManagementSettings: {
+        keyExportable: true
+        observedCertificates: [
+          ''
```

Comment:
Intended?

Reply:
For now yes. I have to check what kind of input is requested here..

Reply:
Hey @johnlokerse, based on your latest comments, would you agree that it may make sense to maybe not set the value in the test at all if it's empty? Instead, a default in the module may be good enough so a user could overwrite it if they'd really want to, but doesn't need to provide it.

Reply:
Hey @AlexanderSehr, yes good plan. I have set the default value for the `osProfile` parameter including an empty `observedCertificates` property.